agnosticv-operator/operator/agnosticvrepo.py

```diff
@@ -0,0 +1,386 @@
+
+        if self.last_successful_git_hexsha:
+            git_diff_output = self.git_repo.git.diff(
+                self.last_successful_git_hexsha, name_only=True
+            )
+            if git_diff_output:
+                self.git_changed_files = git_diff_output.split("\n")
+        else:
+            self.git_changed_files = []
+
+    def __git_repo_clone(self, logger):
+        self.git_repo = git.Repo.clone_from(
+            env = self.git_env,
+            to_path = self.git_repo_path,
+            url = self.git_url,
+        )
+        self.__git_repo_checkout(logger=logger)
+
+    def __git_repo_pull(self, logger):
+        if not self.git_repo:
+            self.git_repo = git.Repo(self.git_repo_path)
+        with self.git_repo.git.custom_environment(**self.git_env):
+            self.git_repo.remotes.origin.fetch()
+        self.__git_repo_checkout(logger=logger)
+
+    def validate_component_definition(self, path, definition):
+        self.validate_execution_environment(
+            path = path,
+            value = definition['__meta__'].get('deployer', {}).get('execution_environment')
+        )
+
+    def validate_execution_environment(self, path, value):
+        if not value:
+            return
+        if 'name' in value:
+            if 'image' in value:
+                raise AgnosticVComponentValidationError(
+                    f"{path}: Execution environment cannot specify both name and image"
+                )
+            if 'pull' in value:
+                raise AgnosticVComponentValidationError(
+                    f"{path}: Execution environment cannot specify both name and pull"
+                )
+            for allow in self.execution_environment_allow_list:
+                if 'name' in allow and re.search(allow['name'], value['name']):
+                    return
+            raise AgnosticVComponentValidationError(f"{path}: Execution environment name not allowed: {value['name']}")
+        elif 'image' in value:
+            for allow in self.execution_environment_allow_list:
+                if 'image' in allow and re.search(allow['image'], value['image']):
+                    return
+            raise AgnosticVComponentValidationError(f"{path}: Execution environment image not allowed: {value['image']}")
+        else:
+            raise AgnosticVComponentValidationError(f"{path}: Execution environment requires name or image")
+
+    async def delete_components(self, logger):
+        async for agnosticv_component in AgnosticVComponent.list(
+            label_selector = f"{Babylon.agnosticv_repo_label}={self.name}",
+            namespace = self.namespace,
+        ):
+            logger.info(f"Propagating delete of {self} to {agnosticv_component}")
+            await agnosticv_component.delete()
+
+    async def get_component_definition(self, path, logger):
+        proc = await asyncio.create_subprocess_shell(
+            f"cd {self.agnosticv_path} && {agnosticv_cli_path} --merge {path}",
```

Comment:
FYI alternatively to yaml, you can use `--output json` and load the dict using the json library.